File: ch30p814takeAPic2/p681takeAPic2/SecondViewController.m


#import "SecondViewController.h"
#import "ViewController.h"

@implementation SecondViewController {
    UIImage* image;
    __weak IBOutlet UIImageView* iv;
}

- (id)initWithNibName:(NSString *)nibNameOrNil bundle:(NSBundle *)nibBundleOrNil image:(UIImage*)im
{
    self = [super initWithNibName:nibNameOrNil bundle:nibBundleOrNil];
    if (self) {
        self.title = @"Decide";
        self.navigationItem.rightBarButtonItem = [[UIBarButtonItem alloc] initWithTitle:@"Use" style:UIBarButtonItemStyleBordered target:self action:@selector(doUse:)];
        self->image = im;
    }
    return self;
}

- (void) viewWillAppear:(BOOL)animated {
    [super viewWillAppear: animated];
    [self.navigationController setNavigationBarHidden:NO animated:NO];
    self->iv.image = image;
}

- (void) viewWillDisappear:(BOOL)animated {
    [super viewWillDisappear:animated];
    [self.navigationController setNavigationBarHidden:YES animated:YES];
}

- (void) doUse: (id) sender {
    ViewController* vc = (id)self.presentingViewController;
    [vc doUse: self->image];
}


@end

File: iOS13bookExamples/bk2ch17p704takeAPicture2/ch30p962takeAPicture2/SecondViewController.swift


import UIKit

class SecondViewController: UIViewController {

    var image : UIImage!
    @IBOutlet var iv : UIImageView!
    
    init(image im:UIImage!) {
        self.image = im
        super.init(nibName: "SecondViewController", bundle: nil)
        self.title = "Decide"
        self.navigationItem.rightBarButtonItem = UIBarButtonItem(title: "Use", style: .plain, target: self, action: #selector(doUse))
    }
    
    required init(coder: NSCoder) {
        fatalError("NSCoding not supported")
    }
    
    override func viewWillAppear(_ animated: Bool) {
        super.viewWillAppear(animated)
        self.navigationController?.setNavigationBarHidden(false, animated: false)
        self.iv.image = self.image
    }
    
    override func viewWillDisappear(_ animated: Bool) {
        super.viewWillDisappear(animated)
        self.navigationController?.setNavigationBarHidden(true, animated: true)
    }
    
    @objc func doUse(_ sender: Any) {
        let vc = self.presentingViewController as! ViewController
        vc.doUse(self.image)
    }
}
